File: packages/shared/types/mealPlan.ts
import { MealType } from '../enums/mealType'

export interface IMealPlan {
  id: string
  projectId: string
  date: string
  recipeName: string
  recipeId?: string
  mealType?: MealType
  imagePath?: string
  createdAt: string
  updatedAt: string
  visibility?: 'private'
  ownerId?: string
}

File: packages/ios/Kairos/Kairos/Models/MealPlan.swift
import Foundation

/// Mirrors `packages/shared/types/mealPlan.ts`.
struct MealPlan: Codable, Identifiable, Sendable, Equatable {
    let id: String
    let projectId: String
    /// ISO date string.
    var date: String
    var recipeName: String
    var recipeId: String?
    var mealType: MealType?
    var imagePath: String?
    let createdAt: String
    var updatedAt: String
    var visibility: String?
    var ownerId: String?
}
